ghedesigner/radial_numerical_borehole.py

```diff
@@ -373,67 +332,30 @@ def fill_f2(fx_2, cell):
             # https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.lapack.dgtsv.html#scipy.linalg.lapack.dgtsv
             dgtsv(_dl, _d, _du, _b, overwrite_b=1)  # TODO: Do we really need lapack just to do a TDMA solution?
 
-            radial_cell[previous_temperature_idx, :] = _b
-            radial_cell[temperature_idx, :] = _b
-
-            if calculate_at_bh_wall:
-                raise ValueError(
-                    "This portion of the code does not currently run with this "
-                    "vectorized radial numerical implementation."
-                )
-                # bh_wall_temp = 0
-                #
-                # # calculate bh wall temp
-                # for idx, _ in enumerate(self.cells):
-                #     west_cell = self.cells[idx]
-                #     east_cell = self.cells[idx + 1]
-                #
-                #     if (
-                #         west_cell.type == RadialCellType.GROUT
-                #         and east_cell.type == RadialCellType.SOIL
-                #     ):
-                #         west_conductance_num = 2 * pi * west_cell.conductivity
-                #         west_conductance_den = log(
-                #             west_cell.outer_radius / west_cell.inner_radius
-                #         )
-                #         west_conductance = west_conductance_num / west_conductance_den
-                #
-                #         east_conductance_num = 2 * pi * east_cell.conductivity
-                #         east_conductance_den = log(
-                #             east_cell.center_radius / west_cell.inner_radius
-                #         )
-                #         east_conductance = east_conductance_num / east_conductance_den
-                #
-                #         bh_wall_temp_num_1 = west_conductance * west_cell.temperature
-                #         bh_wall_temp_num_2 = east_conductance * east_cell.temperature
-                #         bh_wall_temp_num = bh_wall_temp_num_1 + bh_wall_temp_num_2
-                #         bh_wall_temp_den = west_conductance + east_conductance
-                #         bh_wall_temp = bh_wall_temp_num / bh_wall_temp_den
-                #
-                #         break
-                #
-                # g.append(self.c_0 * ((bh_wall_temp - init_temp) / heat_flux))
-            else:
-                g.append(
-                    self.c_0
-                    * ((radial_cell[temperature_idx, 0] - init_temp) / heat_flux - resist_bh_effective)
-                )
+            radial_cell[CellProps.TEMP, :] = _b
+
+            # compute standard g-functions
+            g.append(self.c_0 * ((radial_cell[CellProps.TEMP, 0] - init_temp) / heat_flux - resist_bh_effective))
+
+            # compute g-functions at bh wall
+            bh_wall_temp = radial_cell[CellProps.TEMP, self.bh_wall_idx]
+            g_bhw.append(self.c_0 * ((bh_wall_temp - init_temp) / heat_flux))
```

Comment:
Adding back this capability means evaluating the temperature rise at the BHW instead of the center of the borehole, and this can happen during each iteration along with the typical temperature rise evaluation.